src/crush/CrushCompiler.cc

```diff
@@ -361,7 +361,15 @@ int CrushCompiler::parse_device(iter_t const& i)
   item_id[name] = id;
   id_item[id] = name;
 
-  if (verbose) err << "device " << id << " '" << name << "'" << std::endl;
+  if (verbose) err << "device " << id << " '" << name << "'";
+
+  if (i->children.size() > 3) {
+    string c = string_node(i->children[4]);
+    crush.set_item_class(id, c);
+    if (verbose) err << " class " << " '" << c << "'" << std::endl;
+  } else {
+    if (verbose) err << std::endl;
+  }
```

Comment:
maybe add the decompile case to this patch too, so that a decompile -> compile cycle will not lose informaiton?